saleor/static/dashboard-next/products/components/ProductVariantPage/ProductVariantPage.tsx

```diff
@@ -0,0 +1,265 @@
+  placeholderImage?: string;
+  onBack();
+  onDelete();
+  onSubmit(data: any);
+  onImageSelect(images: string[]);
+  onVariantClick(variantId: string);
+}
+
+const decorate = withStyles(theme => ({
+  root: {
+    "& input": {
+      width: "100%"
+    },
+    display: "grid",
+    gridGap: `${theme.spacing.unit * 2}px`,
+    gridTemplateColumns: "1fr 2fr",
+    [theme.breakpoints.down("sm")]: {
+      gridGap: `${theme.spacing.unit}px`,
+      gridTemplateColumns: "1fr"
+    }
+  }
+}));
+
+const ProductVariantPage = decorate<ProductVariantPageProps>(
+  ({
+    classes,
+    errors: formErrors,
+    loading,
+    placeholderImage,
+    saveButtonBarState,
+    variant,
+    onBack,
+    onDelete,
+    onImageSelect,
+    onSubmit,
+    onVariantClick
+  }) => {
+    const attributes = variant
+      ? variant.attributes.reduce((prev, curr) => {
+          prev[curr.attribute.slug] = curr.value.slug;
+          return prev;
+        }, {})
+      : {};
+    const variantImages = variant
+      ? variant.images.edges.map(edge => edge.node.id)
+      : [];
+    const productImages = variant
+      ? variant.product.images.edges
+          .map(edge => edge.node)
+          .sort((prev, next) => (prev.sortOrder > next.sortOrder ? 1 : -1))
+      : undefined;
+    const images = productImages
+      ? productImages
+          .filter(image => variantImages.indexOf(image.id) !== -1)
+          .sort((prev, next) => (prev.sortOrder > next.sortOrder ? 1 : -1))
+      : undefined;
+    const handleImageSelect = (images: string[]) => {};
+    return (
+      <Toggle>
+        {(isModalActive, { toggle: toggleDeleteModal }) => (
+          <Toggle>
+            {(isImageSelectModalActive, { toggle: toggleImageSelectModal }) => (
+              <>
+                <Container width="md">
+                  <PageHeader
+                    title={variant ? variant.name : undefined}
+                    onBack={onBack}
+                  >
+                    <IconButton onClick={toggleDeleteModal} disabled={loading}>
+                      <DeleteIcon />
+                    </IconButton>
+                  </PageHeader>
+                  <Form
+                    initial={{
+                      costPrice:
+                        variant && variant.costPrice
+                          ? variant.costPrice.amount.toString()
+                          : null,
+                      priceOverride:
+                        variant && variant.priceOverride
+                          ? variant.priceOverride.amount.toString()
+                          : null,
+                      sku: variant && variant.sku,
+                      stock:
+                        variant && variant.quantity ? variant.quantity : "",
+                      ...attributes
+                    }}
+                    errors={formErrors}
+                    onSubmit={onSubmit}
+                    key={variant ? CRC.str(JSON.stringify(variant)) : "loading"}
```

Comment:
Why do we use `JSON.stringify()` in some places and `CRC.str()` in others?

Reply:
#2546